x-pack/plugins/security/server/authentication/providers/kerberos.ts

```diff
@@ -332,11 +335,14 @@ export class KerberosAuthenticationProvider extends BaseAuthenticationProvider {
    * Extracts `Negotiate` challenge from the list of challenges returned with Elasticsearch error if any.
    * @param error Error to extract challenges from.
    */
-  private getNegotiateChallenge(error: LegacyElasticsearchError) {
+  private getNegotiateChallenge(error: errors.ResponseError) {
+    // We extract headers from the original Elasticsearch error and not from the top-level `headers`
+    // property of the Elasticsearch client error since client merges multiple `WWW-Authenticate`
+    // headers into one using comma as a separator. That makes it hard to correctly parse the header
+    // since `WWW-Authenticate` values can also include commas.
     const challenges = ([] as string[]).concat(
-      (error.output.headers as { [key: string]: string })[WWWAuthenticateHeaderName]
+      error.body?.error?.header?.[WWWAuthenticateHeaderName] || []
```

Comment:
note: I didn't test this as part of my review, but I will if you'd like me to.

Reply:
No worries, I specifically checked this. We have integration tests that test this behavior and they failed when I initially tried to use `error.headers`.